### PR TITLE
expression: migrate test-infra to testify for typeinfer_test.go

### LINKED ISSUE #29143: migrate test-infra to testify for `expression/typeinfer_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #29143

Problem Summary:

migrate test-infra to testify for expression/typeinfer_test.go

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: expression/typeinfer_test.go [hunk 1]
```diff
@@ -42,125 +42,17 @@ type typeInferTestCase struct {
 	decimal int
 }
 
-type testInferTypeSuite struct {
+type InferTypeSuite struct {
```
Comment (tisonkun) [inline expression/typeinfer_test.go:45]: I guess you can just inline the Parser instance where it is required.

Comment (hawkingrei) [reply expression/typeinfer_test.go]: I have removed it.

**Comment by hawkingrei** [comment]
@tisonkun @zimulala @AilinKid PTAL

**Review by xiongjiwei — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Pull request merged**